pandas/core/common.py

```diff
@@ -2847,11 +2847,11 @@ def _get_handle(path, mode, encoding=None, compression=None):
 
         if compression == 'gzip':
             import gzip
-            f = gzip.GzipFile(path, 'rb')
+            f = gzip.GzipFile(path, mode)
```

Comment:
[as far as I cant tell](https://docs.python.org/2/library/gzip.html), if we don't pass `mode` then it will default to `rb` and then `to_csv` will fail.